clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -0,0 +1,1005 @@
+     * @param <K>                               The key type.
+     * @param <V>                               The value type.
+     * @return                                  The list value.
+     */
+    static <K, V> List<V> getOrCreateListValue(Map<K, List<V>> map, K key) {
+        List<V> list = map.get(key);
+        if (list != null)
+            return list;
+        list = new LinkedList<V>();
+        map.put(key, list);
+        return list;
+    }
+
+    /**
+     * Send an exception to every element in a collection of KafkaFutures.
+     *
+     * @param futures                           The collection of KafkaFuture objects.
+     * @param exc                               The exception
+     * @param <T>                               The KafkaFuture result type.
+     */
+    private static <T> void completeAllExceptionally(Collection<KafkaFuture<T>> futures, Throwable exc) {
+        for (KafkaFuture<?> future : futures) {
+            future.completeExceptionally(exc);
+        }
+    }
+
+    /**
+     * Get the current time remaining before a deadline as an integer.
+     *
+     * @param now                               The current time in milliseconds.
+     * @param deadlineMs                        The deadline time in milliseconds.
+     * @return                                  The time delta in milliseconds.
+     */
+    static int getTimeoutMsRemainingAsInt(long now, long deadlineMs) {
+        long deltaMs = deadlineMs - now;
+        if (deltaMs > Integer.MAX_VALUE)
+            deltaMs = Integer.MAX_VALUE;
+        else if (deltaMs < Integer.MIN_VALUE)
+            deltaMs = Integer.MIN_VALUE;
+        return (int) deltaMs;
+    }
+
+    /**
+     * Generate the client id based on the configuration.
+     *
+     * @param config                            The configuration
+     *
+     * @return                                  The client id
+     */
+    static String generateClientId(AdminClientConfig config) {
+        String clientId = config.getString(AdminClientConfig.CLIENT_ID_CONFIG);
+        if (clientId.length() > 0)
+            return clientId;
+        return "adminclient-" + ADMIN_CLIENT_ID_SEQUENCE.getAndIncrement();
+    }
+
+    /**
+     * Get the deadline for a particular call.
+     *
+     * @param optionTimeoutMs                   The timeout option given by the user.
+     *
+     * @return                                  The deadline in milliseconds.
+     */
+    private long getDeadlineMs(int optionTimeoutMs) {
+        if (optionTimeoutMs >= 0)
+            return time.milliseconds() + optionTimeoutMs;
+        // If optionTimeoutMs is less than 0, we use the default timeout.
+        return time.milliseconds() + defaultTimeoutMs;
+    }
+
+    /**
+     * Pretty-print an exception.
+     *
+     * @param throwable                         The exception.
+     *
+     * @return                                  A compact human-readable string.
```

Comment:
Yes. There is huge amounts of whitespace. I suspect it's that way because there's an attempt at having alignment across all methods. But the value of that doesn't seem worth it.